jenkins/pipeline-parallel-mtr.groovy

```diff
@@ -606,8 +621,97 @@ pipeline {
                         git branch: JENKINS_SCRIPTS_BRANCH, url: JENKINS_SCRIPTS_REPO
 
                         checkoutSources()
+
+                        script {
+                            // Set BUILD_PARAMS_TYPE based on ANALYZER_OPTS
+                            if (env.ANALYZER_OPTS) {
+                                if (env.ANALYZER_OPTS.contains('ASAN')) {
+                                    env.BUILD_PARAMS_TYPE = 'asan'
+                                } else if (env.ANALYZER_OPTS.contains('VALGRIND')) {
+                                    env.BUILD_PARAMS_TYPE = 'valgrind'
+                                } else if (env.ANALYZER_OPTS.contains('UBSAN')) {
+                                    env.BUILD_PARAMS_TYPE = 'ubsan'
+                                } else if (env.ANALYZER_OPTS.contains('MSAN')) {
+                                    env.BUILD_PARAMS_TYPE = 'msan'
+                                } else {
+                                    env.BUILD_PARAMS_TYPE = 'standard'
+                                }
+                            } else {
+                                env.BUILD_PARAMS_TYPE = 'standard'
+                            }
+
+                            // Extract MySQL version from sources
+                            def versionFile = sh(returnStdout: true, script: """
+                                if [ -f sources/VERSION ]; then
+                                    cat sources/VERSION
+                                elif [ -f sources/MYSQL_VERSION ]; then
+                                    cat sources/MYSQL_VERSION
+                                fi
+                            """).trim()
+
+                            if (versionFile) {
+                                def mysqlVersion = sh(returnStdout: true, script: """
+                                    echo "${versionFile}" | grep -E '^MYSQL_VERSION_MAJOR|^MYSQL_VERSION_MINOR|^MYSQL_VERSION_PATCH' | cut -d= -f2 | paste -sd'.' -
+                                """).trim()
+                                env.MYSQL_VERSION = mysqlVersion
```

Comment:
All of it is already doe with:
```
SERVER_VERSION = validatePsBranch()
```
at https://github.com/Percona-Lab/ps-build/blob/8.0/jenkins/pipeline-parallel-mtr.groovy#L585C28-L585C56
I'm using `SERVER_VERSION` because it's either version of MySQL or Percona Server for MySQL.